frontend/public/components/monitoring/alerting.tsx

```diff
@@ -582,183 +585,239 @@ const alertStateToProps = (state: RootState, { match }): AlertsDetailsPageProps
+                  </Annotation>
+                )}
+              </dl>
+            </div>
+            <div className="col-sm-6">
+              <dl className="co-m-pane__details">
+                <dt>
+                  <PopoverField label={t('public~Source')} body={sourceHelp} />
+                </dt>
+                <dd>{alert && _.startCase(alertSource(alert))}</dd>
+                <dt>
+                  <PopoverField label={t('public~State')} body={alertStateHelp} />
+                </dt>
+                <dd>
+                  <AlertState state={state} />
+                  <AlertStateDescription alert={alert} />
+                </dd>
+              </dl>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-xs-12">
+              <dl className="co-m-pane__details" data-test="label-list">
+                <dt>{t('public~Labels')}</dt>
+                <dd>
+                  {_.isEmpty(labels) ? (
+                    <div className="text-muted">No labels</div>
+                  ) : (
+                    <div className={`co-text-${AlertResource.kind.toLowerCase()}`}>
+                      {_.map(labels, (v, k) => (
+                        <Label key={k} k={k} v={v} />
+                      ))}
+                    </div>
+                  )}
+                </dd>
+              </dl>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-xs-12">
+              <dl className="co-m-pane__details">
+                <dt>{t('public~Alerting rule')}</dt>
+                <dd>
+                  <div className="co-resource-item">
+                    <MonitoringResourceIcon resource={RuleResource} />
+                    <Link
+                      to={
+                        namespace
+                          ? `/dev-monitoring/ns/${namespace}/rules/${rule?.id}`
+                          : ruleURL(rule)
+                      }
+                      data-test="alert-rules-detail-resource-link"
+                      className="co-resource-item__resource-name"
+                    >
+                      {_.get(rule, 'name')}
+                    </Link>
+                  </div>
+                </dd>
+              </dl>
+            </div>
+          </div>
+        </div>
+      </div>
+      {silencesLoaded && !_.isEmpty(alert?.silencedBy) && (
+        <div className="co-m-pane__body">
+          <div className="co-m-pane__body-group">
+            <SectionHeading text="Silenced By" />
+            <div className="row">
+              <div className="col-xs-12">
+                <Table
+                  aria-label="Silenced By"
+                  data={alert?.silencedBy}
+                  Header={silencesTableHeader}
+                  loaded={true}
+                  Row={SilenceTableRow}
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      )}
+    </>
+  );
+};
+
```

Comment:
It will good if we can extract out this component into a separate file as it will make this file a bit smaller and more readable.